core/src/main/java/com/linecorp/armeria/internal/common/Http2GoAwayHandler.java

```diff
@@ -77,6 +77,14 @@ public void onGoAwayReceived(Channel channel, int lastStreamId, long errorCode,
 
         onGoAway(channel, "Received", lastStreamId, errorCode, debugData);
 
+        if (lastStreamId == Integer.MAX_VALUE && errorCode == Http2Error.NO_ERROR.code()) {
+            // Should not close a connection if a last stream identifier is 2^31-1 and a NO_ERROR code.
+            // A server attempted to gracefully shut down a connection.
```

Comment:
suggestion:
- A server -> "The server" , because there's only 1 server per connection
- shut down -> "shutdown a connection" or "shut a connection down"